src/systems/systems/src/hydraulic/linear_actuator.rs

```diff
@@ -0,0 +1,712 @@
+        connected_body: &LinearActuatedRigidBodyOnHingeAxis,
+        context: &UpdateContext,
+    ) {
+        self.last_position = self.position;
+        self.position = connected_body.linear_extension_to_anchor();
+
+        self.position_normalized = (self.position - self.min_absolute_length) / self.total_travel;
+
+        self.delta_displacement = self.position - self.last_position;
+
+        self.speed = self.delta_displacement / context.delta_as_time();
+    }
+
+    fn update_fluid_displacements(&mut self, context: &UpdateContext) {
+        let mut volume_to_actuator = Volume::new::<cubic_meter>(0.);
+        let mut volume_to_reservoir = Volume::new::<cubic_meter>(0.);
+
+        if self.delta_displacement > Length::new::<meter>(0.) {
+            volume_to_actuator = self.delta_displacement * self.bore_side_area;
+            volume_to_reservoir = volume_to_actuator / self.volume_extension_ratio;
+        } else if self.delta_displacement < Length::new::<meter>(0.) {
+            volume_to_actuator = -self.delta_displacement * self.rod_side_area;
+            volume_to_reservoir = volume_to_actuator * self.volume_extension_ratio;
+        }
+
+        if self.delta_displacement >= Length::new::<meter>(0.) {
+            self.signed_flow = volume_to_actuator / context.delta_as_time();
+        } else {
+            self.signed_flow = -volume_to_actuator / context.delta_as_time();
+        }
+
+        self.volume_to_actuator_accumulator += volume_to_actuator;
+        self.volume_to_res_accumulator += volume_to_reservoir;
+    }
+
+    fn update_force(&mut self, hydraulic_pressure: Pressure) {
+        if self.mode == LinearActuatorMode::ClosedValves {
+            let position_error = self.closed_valves_reference_position - self.position_normalized;
+            self.force = Force::new::<newton>(
+                position_error.get::<ratio>() * self.fluid_compression_spring_constant
+                    - self.speed.get::<meter_per_second>()
+                        * self.fluid_compression_damping_constant,
+            );
+        } else if self.mode == LinearActuatorMode::Damping {
+            self.force = Force::new::<newton>(
+                -self.speed.get::<meter_per_second>() * self.active_hydraulic_damping_constant,
+            );
+        } else {
+            self.compute_control_force(hydraulic_pressure);
+        }
+    }
+
+    fn close_control_valves(&mut self) {
+        if self.mode != LinearActuatorMode::ClosedValves {
+            self.closed_valves_reference_position = self.position_normalized;
+            self.mode = LinearActuatorMode::ClosedValves;
+        }
+    }
+
+    fn open_control_valves_for_position_control(&mut self) {
+        self.mode = LinearActuatorMode::PositionControl;
+    }
+
+    fn set_position_target(&mut self, target_position: Ratio) {
+        self.opened_valves_target_position = target_position;
+    }
+
+    fn compute_control_force(&mut self, hydraulic_pressure: Pressure) {
```

Comment:
Need to have a closer look on that implementation, not sure how it would look with that Enum/structs !?

For the modes, those are actually functional modes of the actuator implying different way of computing forces. This is basically an abstraction of what would happen with internal solenoids and super low level hydraulic sorcery we can't even imagine simulate.
There is a state machine but it is at the charge of the actuator user. The actuator just receives current mode through trait HydraulicAssemblyController::requested_mode().

For example an aileron controller would implement HydraulicAssemblyController::requested_mode(), and will send position control if there is pressure and computer is alive, else damping mode.